src/elements/mod.rs

```diff
@@ -261,9 +255,33 @@ pub fn serialize<T: Serialize>(val: T) -> Result<Vec<u8>, T::Error> {
 	Ok(buf)
 }
 
+/// Deserialize module from file.
+#[cfg(feature = "std")]
+pub fn deserialize_file<P: AsRef<::std::path::Path>>(p: P) -> Result<Module, Error> {
+	use std::io::Read;
+
+	let mut contents = Vec::new();
+
+	::std::fs::File::open(p)
```

Comment:
why don't you implement `io::Read` for `std::io::Read` instead of doubling memory requirements both here and below in `serialize_to_file` ?

Reply:
Actually, concern is about `serialize_to_file` only :)

Reply:
Fixed for both `serialize_to_file` and `deserialize_file`!